lib/ansible/modules/system/ufw.py

```diff
@@ -275,131 +275,208 @@
+                has_ipv6 = any([ipv6 for (no, ipv6) in lines])
+                if relative_to_cmd == 'first-ipv4':
+                    relative_to = 1
+                elif relative_to_cmd == 'last-ipv4':
+                    relative_to = max([no for (no, ipv6) in lines if not ipv6]) if has_ipv4 else 1
+                elif relative_to_cmd == 'first-ipv6':
+                    relative_to = max([no for (no, ipv6) in lines if not ipv6]) + 1 if has_ipv4 else 1
+                elif relative_to_cmd == 'last-ipv6':
+                    relative_to = last_number if has_ipv6 else last_number + 1
+                insert_to = self.params['insert'] + relative_to
+                if insert_to > last_number:
+                    # ufw does not like it when the insert number is larger than the
+                    # maximal rule number for IPv4/IPv6.
+                    insert_to = None
+            cmd.append([insert_to is not None, "insert %s" % insert_to])
+
+        cmd.append([value])
+        cmd.append([self.params['direction'], "%s" % self.params['direction']])
+        cmd.append([self.params['interface'], "on %s" % self.params['interface']])
+        cmd.append([self.module.boolean(self.params['log']), 'log'])
+
+        for (key, template) in [('from_ip', "from %s"), ('from_port', "port %s"),
+                                ('to_ip', "to %s"), ('to_port', "port %s"),
+                                ('proto', "proto %s"), ('app', "app '%s'")]:
+            value = self.params[key]
+            cmd.append([value, template % (value)])
+
+        ufw_major, ufw_minor, _ = self.ufw_version()
+        # comment is supported only in ufw version after 0.35
+        if (ufw_major == 0 and ufw_minor >= 35) or ufw_major > 0:
+            cmd.append([self.params['comment'], "comment '%s'" % self.params['comment']])
+
+        rules_dry = self.ufw(cmd)
+
+        if self.is_check_mode():
+
+            nb_skipping_line = len(get_lines_that_contains("Skipping", rules_dry))
+
+            if not (nb_skipping_line > 0 and nb_skipping_line == len(rules_dry.splitlines(True))):
+
+                rules_dry = get_lines_that_start_with("### tuple", rules_dry)
+                # ufw dry-run doesn't send all rules so have to compare ipv4 or ipv6 rules
+                if self.is_starting_by_ipv4(self.params['from_ip']) or self.is_starting_by_ipv4(self.params['to_ip']):
+                    if self.get_lines_that_contains_ipv4(self.pre_rules) != self.get_lines_that_contains_ipv4(rules_dry):
+                        self.__setChanged()
+                elif self.is_starting_by_ipv6(self.params['from_ip']) or self.is_starting_by_ipv6(self.params['to_ip']):
+                    if self.get_lines_that_contains_ipv6(self.pre_rules) != self.get_lines_that_contains_ipv6(rules_dry):
+                        self.__setChanged()
+                elif self.pre_rules != rules_dry:
+                    self.__setChanged()
+
+    def __setChanged(self):
```

Comment:
Is there a reason you use two underscores here (and not for any other method)? Also, does this really needs to be its own method? (Same question for `is_check_mode`, which could also be done by setting `self.check_mode = module.check_mode` in `__init__` and then using `self.check_mode` instead of `self.is_check_mode()`.)